src/cryptonote_core/service_node_list.cpp

```diff
@@ -77,23 +77,21 @@ namespace feature = cryptonote::feature;
 namespace service_nodes {
 static auto logcat = log::Cat("service_nodes");
 
-size_t constexpr STORE_LONG_TERM_STATE_INTERVAL = 10000;
-
 constexpr auto X25519_MAP_PRUNING_INTERVAL = 5min;
 constexpr auto X25519_MAP_PRUNING_LAG = 24h;
 static_assert(
         X25519_MAP_PRUNING_LAG > cryptonote::config::mainnet::config.UPTIME_PROOF_VALIDITY,
         "x25519 map pruning lag is too short!");
 
-static uint64_t short_term_state_cull_height(hf hf_version, uint64_t block_height) {
-    size_t constexpr DEFAULT_SHORT_TERM_STATE_HISTORY = 6 * STATE_CHANGE_TX_LIFETIME_IN_BLOCKS;
-    static_assert(
-            DEFAULT_SHORT_TERM_STATE_HISTORY >= 360,  // Arbitrary, but raises a compilation
-                                                      // failure if it gets shortened.
-            "not enough short term state storage for blink quorum retrieval!");
-    uint64_t result = (block_height < DEFAULT_SHORT_TERM_STATE_HISTORY)
-                            ? 0
-                            : block_height - DEFAULT_SHORT_TERM_STATE_HISTORY;
+static uint64_t min_recent_height(cryptonote::network_type nettype, uint64_t height) {
+    const uint64_t KEEP_WINDOW = cryptonote::get_config(nettype).HISTORY_RECENT_KEEP_WINDOW;
+
+    // NOTE: Arbitrary, but raises a compilation failure if it gets shortened.
+    // 360 is derived via (6 * VOTE_LIFETIME) where VOTE_LIFETIME is 60 blocks,
+    // e.g. Keep atleast the last 360 blocks worth of votes (which is short for
+    // state change TXs in this codebase)
+    assert(KEEP_WINDOW >= 360 && "Not enough recent backups for blink quorum retrieval!");
```

Comment:
Comment still says "compilation failure" from when this was a `static_assert`

Reply:
Fixed